### PR TITLE
[release-0.14] Add configmap support for eventing operator and update the manifest of eventing 

### PR DESCRIPTION
**Pull request by houshengbo**
* Add support of configuring ConfigMaps with CR

* Update the manifest of eventing

Fixes #

## Proposed Changes

*
*
*

**Release Note**

```release-note

```

### CODE REVIEW

File: pkg/reconciler/knativeeventing/common/config_maps.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	eventingv1alpha1 "knative.dev/operator/pkg/apis/operator/v1alpha1"
+)
+
+// ConfigMapTransform updates the ConfigMap with the values specified in operator CR
+func ConfigMapTransform(instance *eventingv1alpha1.KnativeEventing, log *zap.SugaredLogger) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		// Let any config in instance override everything else
+		if u.GetKind() == "ConfigMap" {
+			if data, ok := instance.Spec.Config[u.GetName()[len(`config-`):]]; ok {
```
Comment (Cynocracy) [inline pkg/reconciler/knativeeventing/common/config_maps.go:31]: could stripping off the 'config-' ever lead to collisions?

File: pkg/reconciler/knativeeventing/common/extensions_test.go [hunk 2]
```diff
@@ -61,20 +61,20 @@ func TestTransformers(t *testing.T) {
 
 	results, err := platform.Transformers(kubeclient.Get(ctx), ke, logger)
 	assertEqual(t, err, nil)
-	// By default, there are 4 functions.
-	assertEqual(t, len(results), 4)
+	// By default, there are 5 functions.
+	assertEqual(t, len(results), 5)
```
Comment (Cynocracy) [inline pkg/reconciler/knativeeventing/common/extensions_test.go:65]: This seems a bit fragile, is there any way to test that the functions are doing what we'd expect?

File: pkg/reconciler/knativeeventing/common/config_maps.go (hunk 1 above)
Comment (houshengbo) [reply pkg/reconciler/knativeeventing/common/config_maps.go:31]: As long as the CM name starts with "config-", it should be fine.
Luckily, all CMs in knative start with "config-".

File: pkg/reconciler/knativeeventing/common/extensions_test.go (hunk 2 above)
Comment (houshengbo) [reply pkg/reconciler/knativeeventing/common/extensions_test.go:65]: To me, this only tests the number of functions it calls.
For tests verifying each function, we have test cases for each function.

File: pkg/reconciler/knativeeventing/common/config_maps.go (hunk 1 above)
Comment (Cynocracy) [reply pkg/reconciler/knativeeventing/common/config_maps.go:31]: Alright :) Do we have that constraint documented anywhere?

I'd just like it to be formalized if we're depending on it here.

**Pull request merged**